components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -154,17 +221,273 @@ public void setUp() throws Exception {
                 .add(new OpenIDConnectClaimFilterImpl());
         OpenIDConnectServiceComponentHolder.setRequestObjectService(requestObjectService);
         OAuth2ServiceComponentHolder.setKeyIDProvider(new DefaultKeyIDProviderImpl());
-
     }
 
     @Test
     public void testBuildIDToken() throws Exception {
 
+        String clientId = "dabfba9390aa423f8b04332794d83614";
+        OAuth2AccessTokenRespDTO tokenRespDTO = new OAuth2AccessTokenRespDTO();
+        tokenRespDTO.setAccessToken("2sa9a678f890877856y66e75f605d456");
+        AuthenticatedUser user = getDefaultAuthenticatedUserFederatedUser();
+        OAuthTokenReqMessageContext messageContext = getTokenReqMessageContextForUser(user, clientId);
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                clientId);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getSubject(), "carbon.super");
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getClaim("isk"), "idp");
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getClaim("email"), "email");
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getClaim("username"), "username");
+        Assert.assertNotNull(SignedJWT.parse(idToken).getJWTClaimsSet().getClaim("nbf"));
+        Long expirationTime = ((Date) SignedJWT.parse(idToken).getJWTClaimsSet().getClaim("exp")).getTime();
```

Comment:
Would be much clean if we could declare variable at top for the JWT claim set and get the required claims.

For ex:

JWTClaimsSet claims = SignedJWT.parse(idToken).getJWTClaimsSet();
// Get sub
claims.getSubject()
// Get other claims
claims.getClaim("username")

Reply:
Addressed with 92e89804f1ffeedb2e267fe20e207e794a8dfcbc